### PR TITLE
Add font-palette and animated font-palette

### PR DESCRIPTION
**Pull request by foolip**

### CODE REVIEW

**Comment by foolip** [comment]
@ddbeck I think this is an interesting question of splitting vs merging. In the fullness of time this is definitely one feature, but right now there's a difference in when the different parts were implemented which maps cleanly to both distinct spec concepts, BCD entries and use counters. That's why I decided to split these.

cc @drott @tursunova

File: feature-group-definitions/font-palette-animation.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+name: font-palette animation
+spec: https://drafts.csswg.org/css-fonts-4/#font-palette-prop
+usage_stats: https://chromestatus.com/metrics/css/timeline/animated/709
+compat_features:
+  - css.types.color.palette-mix
```
Comment (ddbeck) [inline feature-group-definitions/font-palette-animation.yml:5]: This doesn't exist and probably should. Maybe file an issue on BCD?

(I'm not stressed about missing keys getting in here and there, if we're committed to tagging in BCD.)

File: feature-group-definitions/font-palette.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+spec:
+  - https://drafts.csswg.org/css-fonts-4/#font-palette-prop
+  - https://drafts.csswg.org/css-fonts-4/#font-palette-values
+usage_stats: https://chromestatus.com/metrics/css/timeline/popularity/709
+compat_features:
+  - css.at-rules.font-palette-values
+  - css.at-rules.font-palette-values.base-palette
+  - css.at-rules.font-palette-values.font-family
+  - css.at-rules.font-palette-values.override-colors
+  - css.properties.font-palette
```
Comment (ddbeck) [inline feature-group-definitions/font-palette.yml:11]: Perhaps this should have `CSSFontPaletteValuesRule` and descendants too?

File: feature-group-definitions/font-palette-animation.yml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+name: font-palette animation
+spec: https://drafts.csswg.org/css-fonts-4/#font-palette-prop
+usage_stats: https://chromestatus.com/metrics/css/timeline/animated/709
```
Comment (ddbeck) [inline feature-group-definitions/font-palette-animation.yml:3]: It's not clear to me that this usage counter belongs here, duplicating the other feature. Can you elaborate on this?

**Review by ddbeck — COMMENTED** [review]
This split seems fine and not terribly ambiguous. Maybe BCD could track the animated-ness of the animation feature? 🤷‍♂️

File: feature-group-definitions/font-palette-animation.yml (hunk 1 above)
Comment (foolip) [reply feature-group-definitions/font-palette-animation.yml:5]: Oh, I added this with the intention to add it to BCD but then forgot. I'll do that before merging this PR.

File: feature-group-definitions/font-palette-animation.yml (hunk 3 above)
Comment (foolip) [reply feature-group-definitions/font-palette-animation.yml:3]: It's not the same URL, this is for when `font-palette` is animated.

File: feature-group-definitions/font-palette-animation.yml (hunk 1 above)
Comment (ddbeck) [reply feature-group-definitions/font-palette-animation.yml:5]: Ah, this was a copy-and-paste failure on my part (I opened "both). Sorry!

File: feature-group-definitions/font-palette-animation.yml (hunk 3 above)
Comment (ddbeck) [reply feature-group-definitions/font-palette-animation.yml:3]: Ah, this was a copy-and-paste failure on my part (I opened "both). Sorry!

File: feature-group-definitions/font-palette.yml (hunk 2 above)
Comment (foolip) [reply feature-group-definitions/font-palette.yml:16]: Done above.

File: feature-group-definitions/font-palette-animation.yml (hunk 1 above)
Comment (foolip) [reply feature-group-definitions/font-palette-animation.yml]: Added in https://github.com/mdn/browser-compat-data/pull/22242 at a different path than I originally though. I'll just merge this and we'll sort out any mismatch when we migrate `compat_features` fully.

**Pull request merged**